cime_config/buildnml

```diff
@@ -616,14 +616,7 @@ def buildnml(case, caseroot, component):
                 major = line[-2] if "MAJOR" in line else major
                 minor = line[-2] if "MINOR" in line else minor
         logger.debug("ESMF version major {} minor {}".format(major, minor))
-        expect(int(major) >= 8, "ESMF version should be 8.1 or newer")
-        if esmf_aware_threading:
-            expect(
-                int(minor) >= 2,
-                "ESMF version should be 8.2.0 or newer when using ESMF_AWARE_THREADING",
-            )
-        else:
-            expect(int(minor) >= 1, "ESMF version should be 8.1.0 or newer")
+        expect(int(major) >= 8 and int(minor) >=4, "ESMF version should be 8.4.1 or newer")
```

Comment:
This logic only works for `int(major) == 8`. EMF 9.1 or 9.3 would fail. I would prefer:
```suggestion
        expect(int(major) >= 8, "ESMF version should be 8.4.1 or newer")
        expect(int(major) > 8 or int(minor) >=4, "ESMF version should be 8.4.1 or newer")
```

Reply:
@gold2718 - the above did not work for ESMF 8.4.1 - so I'm happy to look at this later - but I'd like to get this PR merged.